verilog/analysis/checkers/module_filename_rule.cc

```diff
@@ -73,6 +74,14 @@ void ModuleFilenameRule::Lint(const TextStructureView& text_structure,
   // Find all module declarations.
   auto module_matches = FindAllModuleDeclarations(*tree);
 
+  // Remove nested module declarations
+  module_matches.erase(std::remove_if(module_matches.begin(),
+              module_matches.end(),
+              [](verible::TreeSearchMatch& m) {
+                  return m.context.IsInside(NodeEnum::kModuleDeclaration);
+              }),
+          module_matches.end());
+
```

Comment:
Avoid vector::erase() in this case for its potential O(N^2) cost.
Rewrite as `std::copy_if` or `std::remove_copy_if` using a temporary vector.

Reply:
Done.